### PR TITLE
Hands off `Kokkos::Impl::if_c`

### PR DESCRIPTION
**Pull request by dalg24**
@trilinos/sacado
@trilinos/teuchos
@trilinos/adelus
@trilinos/xpetra
@trilinos/tpetra
@trilinos/stokhos
@trilinos/pytrilinos

## Motivation
It is not legal to reach into the `Kokkos::Impl::` namespace
See https://kokkos.github.io/kokkos-core-wiki/ProgrammingGuide/Compatibility.html

## Testing
Did not actually test

### CODE REVIEW

File: packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp [hunk 1]
```diff
@@ -350,7 +350,7 @@ class View< DataType , Arg1Type , Arg2Type , Arg3Type , Impl::ViewMPVectorInterl
   //------------------------------------
   // Allocation of a managed view with possible alignment padding.
 
-  typedef Impl::if_c< traits::is_managed ,
+  typedef std::conditional< traits::is_managed ,
```
Comment (dalg24) [inline packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp:353]: `if_allocation_constructor` is not actually used anywhere.  If it was up to me I would just remove the definition.

Comment (etphipp) [reply packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp:353]: Fine with me.

**Review by etphipp — DISMISSED** [review]
Looks good to me.

File: packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp (hunk 1 above)
Comment (dalg24) [reply packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp:353]: Want me to remove in this PR?

Comment (dalg24) [reply packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp]: Done

**Review by etphipp — APPROVED** [review]
Thanks!

**Review by bartlettroscoe — APPROVED** [review]
The Teuchos changes look very straightforward.

**Review by jhux2 — DISMISSED** [review]
Xpetra change looks fine, thanks.

**Comment by jhux2** [comment]
@dalg24 Here's the dashboard error, in case you can't access it.
```
In file included from /scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/Trilinos/packages/xpetra[RowMatrix/Xpetra_RowMatrix.hpp:57](https://github.com/trilinos/Trilinos//blob/master/RowMatrix/Xpetra_RowMatrix.hpp#L57),
                 from /scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/Trilinos/packages/xpetra[CrsMatrix/Xpetra_CrsMatrix.hpp:53](https://github.com/trilinos/Trilinos//blob/master/CrsMatrix/Xpetra_CrsMatrix.hpp#L53),
                 from /scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/Trilinos/packages/xpetra[Matrix/Xpetra_CrsMatrixWrap_decl.hpp:59](https://github.com/trilinos/Trilinos//blob/master/Matrix/Xpetra_CrsMatrixWrap_decl.hpp#L59),
                 from /scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/pull_request_test/packages/xpetra[Xpetra_CrsMatrixWrap.hpp:1](https://github.com/trilinos/Trilinos//blob/master/Xpetra_CrsMatrixWrap.hpp#L1),
                 from /scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/Trilinos/packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp:70,
                 from /scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/Trilinos/packages/muelu/test/unit_tests/Zoltan.cpp:47:
/scratch/trilinos/jenkins/ascic164/workspace/Trilinos_PR_gcc-8.3.0/Trilinos/packages/xpetra[Vector/Xpetra_Vector.hpp:88](https://github.com/trilinos/Trilinos//blob/master/Vector/Xpetra_Vector.hpp#L88):6: error: need ‘typename’ before ‘std::conditional<std::is_same<typename Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::dual_view_type::t_dev_um::execution_space::memory_space, typename TargetDeviceType::memory_space>::value, typename Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::dual_view_type::t_dev_um, typename Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::dual_view_type::t_host_um>::type’ because ‘std::conditional<std::is_same<typename Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::dual_view_type::t_dev_um::execution_space::memory_space, typename TargetDeviceType::memory_space>::value, typename Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::dual_view_type::t_dev_um, typename Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::dual_view_type::t_host_um>’ is a dependent scope
      std::conditional<
      ^~~
```

**Comment by dalg24** [comment]
Fixed.  Thank you.  I do not have access.

**Review by jhux2 — APPROVED** [review]
lgtm, thanks

**Comment by jhux2** [comment]
Still waiting on @trilinos/tpetra review.

**Review by brian-kelley — APPROVED** [review]

**Pull request merged**